### ISSUE TITLE
semantic: bind type-only import-equals declarations as type imports

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#22786]
## Summary

Oxc currently binds `import type Foo = require(...)` as a value import. TypeScript treats that alias as type-only, so using `Foo` as a value is invalid.

This causes Oxc’s semantic reference data to resolve value references to a symbol that should only be usable in type space.

## Repro

```ts
import type Foo = require("./m");

Foo;
```

TypeScript reports:

```text
TS1361: 'Foo' cannot be used as a value because it was imported using 'import type'.
```

Oxc semantic currently binds `Foo` with `SymbolFlags::Import`, then resolves the value reference `Foo;` to that import symbol.

Expected behavior:

```text
Foo; // should not resolve as a valid value reference to the type-only import-equals alias
```

## TypeScript behavior

TypeScript tracks type-only alias declarations and rejects value uses of aliases imported with `import type`.

Relevant TypeScript source:

- alias/type-only handling:
  https://github.com/microsoft/TypeScript/blob/f350b52331494b68c90ab02e2b6d0828d2a22a74/src/compiler/checker.ts#L4423-L4429
- invalid type-only alias value-use diagnostic path:
  https://github.com/microsoft/TypeScript/blob/f350b52331494b68c90ab02e2b6d0828d2a22a74/src/compiler/checker.ts#L3352-L3357

## Oxc behavior

`TSImportEqualsDeclaration` carries `import_kind`:

https://github.com/oxc-project/oxc/blob/main/crates/oxc_ast/src/ast/ts.rs#L1688-L1694

But the semantic binder currently ignores it and always passes `false` to `declare_symbol_for_import_specifier`:

https://github.com/oxc-project/oxc/blob/main/crates/oxc_semantic/src/binder.rs#L375-L378

That means `import type Foo = require("./m")` is declared as `SymbolFlags::Import` instead of `SymbolFlags::TypeImport`.

The checker only rejects `import type` when the module reference is not external:

https://github.com/oxc-project/oxc/blob/main/crates/oxc_semantic/src/checker/typescript.rs#L176-L184

So the valid syntax form `import type Foo = require("./m")` ends up with value-import semantics.

## Why this matters

This makes Oxc’s reference/declaration graph disagree with TypeScript and can cause type-only import-equals aliases to appear as runtime value dependencies.
